app/src/test/java/org/mozilla/fenix/utils/ClipboardHandlerTest.kt

```diff
@@ -59,4 +65,31 @@ class ClipboardHandlerTest {
         clipboard.setPrimaryClip(ClipData.newHtmlText("Html", clipboardUrl, clipboardUrl))
         assertEquals(clipboardUrl, clipboardHandler.url)
     }
+
+    @Test
+    fun `text should return firstSafePrimaryClipItemText`() {
+        val safeResult = "safeResult"
+        clipboard.setPrimaryClip(ClipData.newPlainText(clipboardUrl, clipboardText))
+        clipboardHandler = spyk(clipboardHandler)
+        every { clipboardHandler getProperty "firstSafePrimaryClipItemText" } propertyType String::class returns safeResult
+
+        val result = clipboardHandler.text
+
+        verify { clipboardHandler getProperty "firstSafePrimaryClipItemText" }
+        assertEquals(safeResult, result)
+    }
+
+    @Test
+    fun `firstSafePrimaryClipItemText should return the result of SafeUrl#stripUnsafeUrlSchemes`() {
+        mockkObject(SafeUrl)
+        every { SafeUrl.stripUnsafeUrlSchemes(any(), any()) } returns "safeResult"
+        clipboard.setPrimaryClip(ClipData.newHtmlText("Html", clipboardUrl, clipboardUrl))
+
+        val result = clipboardHandler.firstSafePrimaryClipItemText
+
+        verify { SafeUrl.stripUnsafeUrlSchemes(testContext, clipboardUrl) }
+        assertEquals("safeResult", result)
+
+        unmockkObject(SafeUrl)
```

Comment:
nit: If the test ever fails, this wouldn't call and possibly affect other tests. Could move this to `@before/@after` or in a finally block.